Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -68,7 +68,7 @@ internal final class SelectableEventLoop: EventLoop {
     }
 
     /* private but tests */ internal let _selector: NIOPosix.Selector<NIORegistration>
-    private let thread: NIOThread
+    private let thread: ManagedAtomicLazyReference<NIOThread>
```

Comment:
Can we add a comment here why this is a `ManagedAtomicLazyReference` and how the threading is working.

Reply:
Done, please re-review.